components/server/src/user/user-service.ts

```diff
@@ -244,11 +248,12 @@ export class UserService {
      * @param workspace - optional, in which case the default billing account will be checked
      * @returns
      */
-    async checkUsageLimitReached(
-        user: User,
-        workspace?: Pick<Workspace, "projectId">,
-    ): Promise<UsageLimitReachedResult> {
-        const attributionId = await this.getWorkspaceUsageAttributionId(user, workspace?.projectId);
+    async checkUsageLimitReached(user: User, organizationId?: string): Promise<UsageLimitReachedResult> {
+        if (!organizationId && user.additionalData?.isMigratedToTeamOnlyAttribution) {
+            throw new Error("organizationId must be provided for org-only users");
+        }
+        const attributionId: AttributionId =
+            organizationId === undefined ? AttributionId.create(user) : { kind: "team", teamId: organizationId };
```

Comment:
May be worth to either add a function, or an overload for 
```
    export function create(userOrTeam: User | Team): AttributionId {
        if (User.is(userOrTeam)) {
            return { kind: "user", userId: userOrTeam.id };
        } else {
            return { kind: "team", teamId: userOrTeam.id };
        }
    }
```
to keep the attribution ID construction similar to users and teams